core/src/main/java/io/aiven/kafka/tieredstorage/metrics/CaffeineStatsCounter.java

```diff
@@ -219,4 +244,26 @@ public double measure(final MetricConfig config, final long now) {
             return value.sum();
         }
     }
+
+    /**
+     * Implementation of {@link Value} that allows fetching a value from provided {@code Long} {@link Supplier}
+     * to avoid unnecessary calls to {@link Sensor#record()} that under the hood has a synchronized block and affects
+     * performance because of that.
+     *
+     * <p>To be used by metrics that require setting to specific values without increment/decrement steps.
+     *
+     * @see AdderValue
+     */
+    private static class SuppliedValue extends Value {
+        private final Supplier<Long> value;
+
+        SuppliedValue(final Supplier<Long> value) {
+            this.value = value;
+        }
+
+        @Override
+        public double measure(final MetricConfig config, final long now) {
+            return value.get();
+        }
+    }
```

Comment:
I don't think we need these 2 classes, we can have just have one and provide LongAdder::sum as value supplier. Also I would prefer to keep the name of the original class.